### PR TITLE
Be more lenient when stripping frameworks and dSYMs for non fat binaries

### LINKED ISSUE #7196: Issue with thinning dSYM files
* [* ] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

integrate a project with `ArcGIS-Runtime-SDK-iOS` pod

`pod install`

Followed by building of resulting workspace in Xcode for simulator

## What did you expect to happen?

Project to build and run on simulator

## What happened instead?

When building for simulator environment project failed whilst running `[CP] Embed Pod Frameworks`  step. The above pod vendors a binary static framework that includes dSYM files but there are only FAT slices for device architectures not simulator. When building for simulator the script fails when stripping the last device architecture lipo fails.

`fatal error: lipo: -remove's specified would result in an empty fat file`

When building for device rather than simulator the build succeeds since the device slices are not removed. Note that commenting out:

`strip_invalid_archs "$binary"`

For the dSYM files allowed the app to build and run in the simulator.

Can this case be detected and the dSYM copy skipped rather than failing with the above error?

### Stack

```
   CocoaPods : 1.4.0.beta.2
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.6.14
        Host : Mac OS X 10.13.1 (17B48)
       Xcode : 9.1 (9B55)
         Git : git version 2.13.6 (Apple Git-96)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2f034bfbd3c18eef1ea398afa5cf9824c7f858fd
               urbanthings - git@bitbucket.org:urbanthings/podspecs.git @ 8a820e4853f85582b0db6eb5dd876fce61ffe322
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
use_frameworks!

target 'CPBug' do
    pod 'ArcGIS-Runtime-SDK-iOS'
end
```

## Project that demonstrates the issue

A project to reproduce issue available here:
https://github.com/mwoollard/cpbug.git

### PR DESCRIPTION
**Pull request by dnkoutso**
closes #7196, #5854

### CODE REVIEW

**Comment by dnkoutso** [comment]
Will update integration specs

File: lib/cocoapods/generator/embed_frameworks_script.rb [hunk 1]
```diff
@@ -136,10 +145,18 @@ def script
...
             binary="$1"
-            # Get architectures for current file
-            archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | rev)"
+            # Get architectures for current target binary
+            binary_archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | awk '{$1=$1;print}' | rev)"
+            # Intersect them with the architectures we are building for
+            intersected_archs="$(echo ${ARCHS[@]} ${binary_archs[@]} | tr ' ' '\\n' | sort | uniq -d)"
+            # If there are no archs supported by this binary then warn the user
+            if [[ -z "$intersected_archs" ]]; then
+              echo "warning: [CP] Vendored binary '$binary' contains architectures ($binary_archs) none of which match the current build architectures ($ARCHS)."
```
Comment (dnkoutso) [inline lib/cocoapods/generator/embed_frameworks_script.rb]: using `warning:` will actually show a warning in Xcode.

Comment (loveNoodles) [reply lib/cocoapods/generator/embed_frameworks_script.rb]: so in order to avoid the problem i should upgrade cocoapods ?

**Comment by revolter** [comment]
Should this be fixed in `1.4.0`?

**Comment by rmaloney3** [comment]
Is there a way to suppress this warning? We have warnings as errors enabled, and a third-party library that we use does not include a dSYM for x86_64. It's closed source, otherwise we would fix it ourselves. I've filed a bug with them, but they haven't fixed it, so I was hoping that it would be possible to suppress it via CocoaPods somehow.

**Pull request merged**